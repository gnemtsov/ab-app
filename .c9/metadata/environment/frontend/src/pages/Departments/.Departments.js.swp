{"filter":false,"title":".Departments.js.swp","tooltip":"/frontend/src/pages/Departments/.Departments.js.swp","undoManager":{"mark":-1,"position":-1,"stack":[]},"ace":{"folds":[],"scrolltop":1875,"scrollleft":0,"selection":{"start":{"row":0,"column":0},"end":{"row":0,"column":0},"isBackwards":true},"options":{"guessTabSize":true,"useWrapMode":true,"wrapToView":true},"firstLineState":{"row":20,"state":"start","mode":"ace/mode/text"}},"timestamp":1541949402988,"hash":"870bd6fac4f36903b6f7c1f6a4e3d3f75a378ebd"}