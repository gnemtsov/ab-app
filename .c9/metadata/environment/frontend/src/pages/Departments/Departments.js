{"filter":false,"title":"Departments.js","tooltip":"/frontend/src/pages/Departments/Departments.js","undoManager":{"mark":11,"position":11,"stack":[[{"start":{"row":22,"column":20},"end":{"row":22,"column":28},"action":"remove","lines":["conf={{ "],"id":2}],[{"start":{"row":22,"column":37},"end":{"row":22,"column":38},"action":"insert","lines":["="],"id":3},{"start":{"row":22,"column":38},"end":{"row":22,"column":39},"action":"insert","lines":["Х"]}],[{"start":{"row":22,"column":38},"end":{"row":22,"column":39},"action":"remove","lines":["Х"],"id":4}],[{"start":{"row":22,"column":38},"end":{"row":22,"column":39},"action":"insert","lines":["{"],"id":5}],[{"start":{"row":22,"column":39},"end":{"row":22,"column":40},"action":"remove","lines":[":"],"id":6},{"start":{"row":22,"column":39},"end":{"row":22,"column":40},"action":"remove","lines":[" "]}],[{"start":{"row":22,"column":61},"end":{"row":22,"column":62},"action":"insert","lines":["}"],"id":7}],[{"start":{"row":22,"column":62},"end":{"row":23,"column":0},"action":"insert","lines":["",""],"id":8},{"start":{"row":23,"column":0},"end":{"row":23,"column":20},"action":"insert","lines":["                    "]}],[{"start":{"row":23,"column":20},"end":{"row":23,"column":22},"action":"remove","lines":[", "],"id":9}],[{"start":{"row":23,"column":30},"end":{"row":23,"column":31},"action":"insert","lines":["="],"id":10},{"start":{"row":23,"column":31},"end":{"row":23,"column":32},"action":"insert","lines":["{"]}],[{"start":{"row":23,"column":32},"end":{"row":23,"column":33},"action":"remove","lines":[":"],"id":11},{"start":{"row":23,"column":32},"end":{"row":23,"column":33},"action":"remove","lines":[" "]},{"start":{"row":23,"column":32},"end":{"row":23,"column":33},"action":"remove","lines":["t"]}],[{"start":{"row":23,"column":32},"end":{"row":23,"column":33},"action":"insert","lines":["t"],"id":12}],[{"start":{"row":23,"column":36},"end":{"row":23,"column":37},"action":"remove","lines":[" "],"id":13},{"start":{"row":23,"column":36},"end":{"row":23,"column":37},"action":"remove","lines":["}"]}]]},"ace":{"folds":[],"scrolltop":171.7421875,"scrollleft":0,"selection":{"start":{"row":23,"column":37},"end":{"row":23,"column":37},"isBackwards":false},"options":{"guessTabSize":true,"useWrapMode":true,"wrapToView":true},"firstLineState":{"row":14,"state":"start","mode":"ace/mode/javascript"}},"timestamp":1541958186033,"hash":"29873d9ce65e34b82ce3217538d6149a0aff6e15"}