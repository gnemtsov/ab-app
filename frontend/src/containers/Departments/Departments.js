import React, { Component } from 'react';
import { connect } from 'react-redux';

import Aux from '../../hoc/Auxillary/Auxillary';
import Spinner from '../../components/UI/Spinner/Spinner';
import AbTable from 'react-ab-table';
import withErrorHandler from '../../hoc/withErrorHandler/withErrorHandler';
import * as actionTypes from "../../store/actionTypes";
import axios from 'axios';

export class Departments extends Component {
    componentDidMount() {
        this.props.onInitDepartments();
    }

    render() {

        let departments = this.props.error ? <p>Departments can't be loaded!</p> : <Spinner />;

        if (this.props.departments) {
            departments = <AbTable data={this.props.departments} />;
        }

        return (
            <Aux>
                {departments}
            </Aux>
        );
    }
}

const mapStateToProps = state => {
    return {
        departments: state.departments.departments,
        error: state.departments.error,
        isAuthenticated: state.auth.isAuthenticated
    };
}

const mapDispatchToProps = dispatch => {
    return {
        onInitDepartments: () => dispatch({
            type: actionTypes.S_INIT_DEPARTMENTS
        })
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(withErrorHandler(Departments, axios));