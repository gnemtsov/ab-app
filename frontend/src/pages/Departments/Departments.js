import React, { Component } from 'react';
import { connect } from 'react-redux';

import Spinner from '../../components/UI/Spinner/Spinner';
import Table from '../../components/Table/Table';
import * as actionTypes from "../../store/actionTypes";

import classes from './Departments.css';

export class Departments extends Component {	
    componentDidMount() {
        this.props.onInitDepartments();
    }

    render() {
        let departments = this.props.error ? <p>Departments can't be loaded!</p> : <Spinner />;

        if (this.props.departments.departments) {
            departments =
                <Table
                    title="Departments"
                    csvExport={true}
                    conf={{ emptyTableMessage: 'No departments found', selectable: true }}
                    rows={this.props.departments.departments.rows}
                    cols={this.props.departments.departments.cols}
                    filter={this.props.departments.filter}
                    onSetFilter={this.props.onSetFilter} />;
        }

        return (
            <div
                className={classes.TableContainer}>
                {departments}
            </div>
        );
    }
}

const mapStateToProps = state => {
    return {
        departments: state.departments,
        error: state.departments.error,
        isAuthenticated: state.auth.isAuthenticated
    };
}

const mapDispatchToProps = dispatch => {
    return {
        onInitDepartments: () => dispatch({
            type: actionTypes.S_INIT_DEPARTMENTS
        }),
        onSetFilter: (filter) => dispatch({
			type: actionTypes.R_SET_DEPARTMENTS_FILTER,
			filter: filter
		})
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Departments);
