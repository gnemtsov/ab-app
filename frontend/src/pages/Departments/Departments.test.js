import React from 'react';

import { configure, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Departments from './Departments';

configure({ adapter: new Adapter() });

describe('Departments', () => {
    it('TODO', () => {
        expect(true).toBe(true);
    });

    /*const snapshoot = (name, departments, onSetFilter, error) => {
        const wrapper = render(
            <Departments
                departments={departments}
                onSetFilter={onSetFilter}
                error={error} />
        );
        it(name, () => {
            expect(wrapper).toMatchSnapshot();
        });
    }
    
    let departments = {
		departments: {
			cols: [
				{ name: 'name', title: 'Pet name' },
				{ name: 'class', title: 'Animal class' },
				{ name: 'age', title: 'Age' },
				{ name: 'gender', title: 'Gender' }
			],
			rows: [
				{ name: 'Buddy', class: 'Dog', age: 3, gender: 'male' },
				{ name: 'Molly', class: 'Cat', age: 15, gender: 'female' },
				{ name: 'Bonnie', class: 'Cat', age: 2, gender: 'female' },
				{ name: 'Coco', class: 'Parrot', age: 22, gender: 'male' },
				{ name: 'Oscar', class: 'Dog', age: 5, gender: 'male' },
				{ name: 'Max', class: 'Turtle', age: 15, gender: 'male' },
				{ name: 'Jack', class: 'Varan', age: 1, gender: 'male' }
			]
		},
		filter: null
	};
	let onSetFilter = () => {};
	let error = undefined;
	
	snapshoot('With data and no error', departments, onSetFilter, error);
	
	error = true;
	snapshoot('With data and error', departments, onSetFilter, error);
	
	error = false;
	departments.departments = null;
	snapshoot('With no data and no error', departments, onSetFilter, error);
	
	error = true;
	snapshoot('With no data and error', departments, onSetFilter, error);*/
});

