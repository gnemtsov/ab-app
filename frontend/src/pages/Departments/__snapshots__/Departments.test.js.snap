// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Departments Renders correctly when error 1`] = `
<div
  class="TableContainer"
>
  <p>
    Departments can't be loaded!
  </p>
</div>
`;

exports[`Departments Renders correctly when no error 1`] = `
<div
  class="TableContainer"
>
  <h1>
    Departments
  </h1>
  <div
    class="Loader"
  >
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
  </div>
</div>
`;
